File: src/main/java/com/sber/yamlprocessor/model/Process.java
package com.sber.yamlprocessor.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;

import java.util.List;

public class Process {
    
    private String id;
    
    @JsonProperty("context-code")
    private ContextCodes contextCode;
    
    private Boolean disabled = false;
    
    @NotNull(message = "Description is required")
    private String description;
    
    @Valid
    @NotNull(message = "Subprocess is required")
    private List<Subprocess> subprocess;

    public Process() {}

    public Process(String description, List<Subprocess> subprocess) {
        this.description = description;
        this.subprocess = subprocess;
    }

    // Getters and Setters
    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public ContextCodes getContextCode() {
        return contextCode;
    }

    public void setContextCode(ContextCodes contextCode) {
        this.contextCode = contextCode;
    }

    public Boolean getDisabled() {
        return disabled;
    }

    public void setDisabled(Boolean disabled) {
        this.disabled = disabled;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public List<Subprocess> getSubprocess() {
        return subprocess;
    }

    public void setSubprocess(List<Subprocess> subprocess) {
        this.subprocess = subprocess;
    }
}

File: modules/digital-credit-process-configurator/src/main/kotlin/com/sber/yamlprocessor/model/Process.kt
package com.sber.yamlprocessor.model

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.persistence.CascadeType
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.OneToMany
import jakarta.persistence.OneToOne
import jakarta.persistence.OrderColumn
import jakarta.persistence.Table

@Entity
@Table(name = "process")
class Process(
    @OneToOne(optional = true)
    @JoinColumn(name = "process_config_id")
    @JsonIgnore
    var processConfig: ProcessConfig? = null,

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "context_code", referencedColumnName = "code")
    @field:JsonProperty("context-code")
    var contextCode: ContextCodesDictionary? = null,

    @Column(name = "disabled")
    var disabled: Boolean = false,

    @Column(name = "description", nullable = false, length = 4000)
    var description: String = "",

    @Column(name = "node_name", length = 255)
    var nodeName: String? = null,

    @Column(name = "node_comment", length = 4000)
    var nodeComment: String? = null,

    @OneToMany(mappedBy = "process", cascade = [CascadeType.ALL], orphanRemoval = true)
    @OrderColumn(name = "subprocess_sort")
    var subprocess: MutableList<Subprocess> = mutableListOf()
) : BaseEntity()
